source/widgets/ProjectStatus/js/configuration.js

```diff
@@ -182,7 +208,21 @@ function OctopusStatusWidgetConfiguration() {
                                         $environmentDropdown.val(),
                                         $(":selected", $environmentDropdown).text()
                                     );
-                                    refreshProjectsAndEnvironmentsInSpaceDropdowns(settings, queryUri(), authToken, $spaceDropdown.val());
+                                    refreshProjectsAndEnvironmentsInSpaceDropdowns(settings, queryUri(), authToken, $spaceDropdown.val(), $projectFilterInput.val());
+                                });
+
+                                $projectFilterInput.on("change", function () {
```

Comment:
If the project filter changes, refresh all the things (it's not a great UX, I'd rather a debounce, but we're limited in what we can do here)